File: challenge-312/ppentchev/python/src/pwc_312/defs.py
# SPDX-FileCopyrightText: Peter Pentchev <roam@ringlet.net>
# SPDX-License-Identifier: BSD-2-Clause
"""Common definitions for the pwc-312 module."""

from __future__ import annotations

import typing


if typing.TYPE_CHECKING:
    from typing import Final


"""The pwc-312 version string, semver-like."""
VERSION: Final = "0.1.0"

File: challenge-293/ppentchev/rust/src/defs.rs
// SPDX-FileCopyrightText: Peter Pentchev <roam@ringlet.net>
// SPDX-License-Identifier: BSD-2-Clause
//! Common definitions for Peter Pentchev's solutions to PWC 293.

use thiserror::Error;

/// An error that occurred while processing the input data.
#[derive(Debug, Error)]
#[non_exhaustive]
#[allow(clippy::error_impl_error)]
pub enum Error {
    /// Did we really overflow the usize type?
    #[error("Did we really overflow the usize or u64 type?!")]
    ArithmeticOverflow,

    /// Something went really, really wrong...
    #[error("similar-dominos internal error: {0}")]
    Internal(String),
}
